internal/handler/msg_saslstart.go

```diff
@@ -68,18 +68,37 @@ func (h *Handler) saslStart(ctx context.Context, dbName string, document *types.
 		return nil, lazyerrors.Error(err)
 	}
 
+	if h.EnableNewAuth {
+		switch mechanism {
+		case "SCRAM-SHA-1", "SCRAM-SHA-256":
+			var response string
+
+			if response, err = h.saslStartSCRAM(ctx, dbName, mechanism, document); err != nil {
+				return nil, err
+			}
+
+			conninfo.Get(ctx).SetBypassBackendAuth()
+
+			return must.NotFail(types.NewDocument(
+				"conversationId", int32(1),
+				"done", false,
+				"payload", types.Binary{B: []byte(response)},
+			)), nil
+		default:
+			msg := fmt.Sprintf("Unsupported authentication mechanism %q.\n", mechanism) +
+				"See https://docs.ferretdb.io/security/authentication/ for more details."
+			return nil, handlererrors.NewCommandErrorMsgWithArgument(handlererrors.ErrAuthenticationFailed, msg, "mechanism")
+		}
+	}
+
 	switch mechanism {
 	case "PLAIN":
```

Comment:
I don't understand this part. In some other parts of the code, `PLAIN` is removed completely, so if `PLAIN` is called, an "Unknown auth mechanism" error will be returned. But here we still keep it. Does it make sense, if `PLAIN` is not supported in the other pars?

Reply:
In old authentication, we don't have user records, so we don't have anything to update. Still, we allow users to authenticate. https://docs.ferretdb.io/security/authentication/